[Automation] java run fails for commons-lang:commons-lang:2.0

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0" ./gradlew clean javaTest -Pcoordinates="commons-lang:commons-lang:1.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25481363170/job/74766432804
Last 300 lines of the log:
```console
> Task :compileJava NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :testClasses

> Task :test FAILED

ToStringBuilderTest > reflectionToStringReadsDeclaredFields() FAILED
    java.lang.ExceptionInInitializerError
        at org.apache.commons.lang.builder.ToStringStyle$MultiLineToStringStyle.<init>(ToStringStyle.java:2084)
        at org.apache.commons.lang.builder.ToStringStyle$MultiLineToStringStyle.<init>(ToStringStyle.java:2074)
        at org.apache.commons.lang.builder.ToStringStyle.<clinit>(ToStringStyle.java:100)
        at org.apache.commons.lang.builder.ToStringBuilder.<clinit>(ToStringBuilder.java:135)
        at commons_lang.commons_lang.ToStringBuilderTest.reflectionToStringReadsDeclaredFields(ToStringBuilderTest.java:20)

        Caused by:
        java.lang.NumberFormatException: multiple points
            at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1914)
            at java.base/jdk.internal.math.FloatingDecimal.parseFloat(FloatingDecimal.java:122)

            at java.base/java.lang.Float.parseFloat(Float.java:556)
14 tests completed, 1 failed
            at org.apache.commons.lang.SystemUtils.getJavaVersionAsFloat(SystemUtils.java:756)
            at org.apache.commons.lang.SystemUtils.<clinit>(SystemUtils.java:469)
            ... 5 more

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
15 actionable tasks: 5 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for commons-lang:commons-lang:1.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.0 at 2026-05-07T07:13:19Z with exit code 1 after 5s
❌ ERROR: 'native-image run' for 2.0 failed with exit code 1.
SOFT-FAIL[native-image run][2.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0" ./gradlew clean test -Pcoordinates="commons-lang:commons-lang:1.0"]
========================================================================================
 commons-lang:commons-lang:1.0:2.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.0 at 2026-05-07T07:13:19Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0" ./gradlew clean compileTestJava -Pcoordinates="commons-lang:commons-lang:1.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 835ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.0 at 2026-05-07T07:13:22Z with exit code 0 after 3s
========================================================================================
 commons-lang:commons-lang:1.0:2.0 stage java run
========================================================================================
Starting stage 'java run' for 2.0 at 2026-05-07T07:13:22Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0" ./gradlew clean javaTest -Pcoordinates="commons-lang:commons-lang:1.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 846ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources
> Task :testClasses

> Task :test FAILED

ToStringBuilderTest > reflectionToStringReadsDeclaredFields() FAILED
    java.lang.ExceptionInInitializerError
        at org.apache.commons.lang.builder.ToStringStyle$MultiLineToStringStyle.<init>(ToStringStyle.java:2084)
        at org.apache.commons.lang.builder.ToStringStyle$MultiLineToStringStyle.<init>(ToStringStyle.java:2074)
        at org.apache.commons.lang.builder.ToStringStyle.<clinit>(ToStringStyle.java:100)
        at org.apache.commons.lang.builder.ToStringBuilder.<clinit>(ToStringBuilder.java:135)
        at commons_lang.commons_lang.ToStringBuilderTest.reflectionToStringReadsDeclaredFields(ToStringBuilderTest.java:20)

        Caused by:
        java.lang.NumberFormatException: multiple points
            at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1914)
14 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

            at java.base/jdk.internal.math.FloatingDecimal.parseFloat(FloatingDecimal.java:122)
            at java.base/java.lang.Float.parseFloat(Float.java:556)
            at org.apache.commons.lang.SystemUtils.getJavaVersionAsFloat(SystemUtils.java:756)
            at org.apache.commons.lang.SystemUtils.<clinit>(SystemUtils.java:469)
            ... 5 more

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 2.0 at 2026-05-07T07:13:26Z with exit code 1 after 4s
❌ ERROR: 'java run' for 2.0 failed with exit code 1.
FAILED[java run][2.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0" ./gradlew clean javaTest -Pcoordinates="commons-lang:commons-lang:1.0"]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0" ./gradlew clean javaTest -Pcoordinates="commons-lang:commons-lang:1.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25481363170/job/74766432790
Last 300 lines of the log:
```console
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :testClasses

> Task :test FAILED

ToStringBuilderTest > reflectionToStringReadsDeclaredFields() FAILED
    java.lang.ExceptionInInitializerError
        at org.apache.commons.lang.builder.ToStringStyle$MultiLineToStringStyle.<init>(ToStringStyle.java:2084)
        at org.apache.commons.lang.builder.ToStringStyle$MultiLineToStringStyle.<init>(ToStringStyle.java:2074)
        at org.apache.commons.lang.builder.ToStringStyle.<clinit>(ToStringStyle.java:100)
        at org.apache.commons.lang.builder.ToStringBuilder.<clinit>(ToStringBuilder.java:135)
        at commons_lang.commons_lang.ToStringBuilderTest.reflectionToStringReadsDeclaredFields(ToStringBuilderTest.java:20)

        Caused by:
        java.lang.NumberFormatException: multiple points
            at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1914)
            at java.base/jdk.internal.math.FloatingDecimal.parseFloat(FloatingDecimal.java:122)
            at java.base/java.lang.Float.parseFloat(Float.java:556)
            at org.apache.commons.lang.SystemUtils.getJavaVersionAsFloat(SystemUtils.java:756)
            at org.apache.commons.lang.SystemUtils.<clinit>(SystemUtils.java:469)
            ... 5 more

14 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
15 actionable tasks: 5 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for commons-lang:commons-lang:1.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.0 at 2026-05-07T07:13:25Z with exit code 1 after 4s
❌ ERROR: 'native-image run' for 2.0 failed with exit code 1.
SOFT-FAIL[native-image run][2.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0" ./gradlew clean test -Pcoordinates="commons-lang:commons-lang:1.0"]
========================================================================================
 commons-lang:commons-lang:1.0:2.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.0 at 2026-05-07T07:13:25Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0" ./gradlew clean compileTestJava -Pcoordinates="commons-lang:commons-lang:1.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 849ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.0 at 2026-05-07T07:13:28Z with exit code 0 after 3s
========================================================================================
 commons-lang:commons-lang:1.0:2.0 stage java run
========================================================================================
Starting stage 'java run' for 2.0 at 2026-05-07T07:13:28Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0" ./gradlew clean javaTest -Pcoordinates="commons-lang:commons-lang:1.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 867ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :processTestResources
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :testClasses

14 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

> Task :test FAILED

ToStringBuilderTest > reflectionToStringReadsDeclaredFields() FAILED
    java.lang.ExceptionInInitializerError
        at org.apache.commons.lang.builder.ToStringStyle$MultiLineToStringStyle.<init>(ToStringStyle.java:2084)
        at org.apache.commons.lang.builder.ToStringStyle$MultiLineToStringStyle.<init>(ToStringStyle.java:2074)
        at org.apache.commons.lang.builder.ToStringStyle.<clinit>(ToStringStyle.java:100)

        at org.apache.commons.lang.builder.ToStringBuilder.<clinit>(ToStringBuilder.java:135)
        at commons_lang.commons_lang.ToStringBuilderTest.reflectionToStringReadsDeclaredFields(ToStringBuilderTest.java:20)

        Caused by:
        java.lang.NumberFormatException: multiple points
            at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1914)
            at java.base/jdk.internal.math.FloatingDecimal.parseFloat(FloatingDecimal.java:122)
            at java.base/java.lang.Float.parseFloat(Float.java:556)
            at org.apache.commons.lang.SystemUtils.getJavaVersionAsFloat(SystemUtils.java:756)
            at org.apache.commons.lang.SystemUtils.<clinit>(SystemUtils.java:469)
            ... 5 more

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date
BUILD FAILED in 2s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 2.0 at 2026-05-07T07:13:32Z with exit code 1 after 4s
❌ ERROR: 'java run' for 2.0 failed with exit code 1.
FAILED[java run][2.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0" ./gradlew clean javaTest -Pcoordinates="commons-lang:commons-lang:1.0"]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.0" ./gradlew clean javaTest -Pcoordinates="commons-lang:commons-lang:1.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25481363170/job/74766432507
Last 300 lines of the log:
```console
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :jar
> Task :compileTestJava
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :testClasses

> Task :test FAILED

ToStringBuilderTest > reflectionToStringReadsDeclaredFields() FAILED
    java.lang.ExceptionInInitializerError
        at org.apache.commons.lang.builder.ToStringStyle$MultiLineToStringStyle.<init>(ToStringStyle.java:2084)
        at org.apache.commons.lang.builder.ToStringStyle$MultiLineToStringStyle.<init>(ToStringStyle.java:2074)
        at org.apache.commons.lang.builder.ToStringStyle.<clinit>(ToStringStyle.java:100)
        at org.apache.commons.lang.builder.ToStringBuilder.<clinit>(ToStringBuilder.java:135)
        at commons_lang.commons_lang.ToStringBuilderTest.reflectionToStringReadsDeclaredFields(ToStringBuilderTest.java:20)

        Caused by:
        java.lang.NumberFormatException: multiple points
            at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1914)
            at java.base/jdk.internal.math.FloatingDecimal.parseFloat(FloatingDecimal.java:122)
            at java.base/java.lang.Float.parseFloat(Float.java:556)
            at org.apache.commons.lang.SystemUtils.getJavaVersionAsFloat(SystemUtils.java:756)
            at org.apache.commons.lang.SystemUtils.<clinit>(SystemUtils.java:469)
            ... 5 more

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

14 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
15 actionable tasks: 5 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for commons-lang:commons-lang:1.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.0 at 2026-05-07T07:13:16Z with exit code 1 after 5s
❌ ERROR: 'native-image run' for 2.0 failed with exit code 1.
SOFT-FAIL[native-image run][2.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.0" ./gradlew clean test -Pcoordinates="commons-lang:commons-lang:1.0"]
========================================================================================
 commons-lang:commons-lang:1.0:2.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.0 at 2026-05-07T07:13:16Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.0" ./gradlew clean compileTestJava -Pcoordinates="commons-lang:commons-lang:1.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 784ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 989ms
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.0 at 2026-05-07T07:13:18Z with exit code 0 after 2s
========================================================================================
 commons-lang:commons-lang:1.0:2.0 stage java run
========================================================================================
Starting stage 'java run' for 2.0 at 2026-05-07T07:13:18Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.0" ./gradlew clean javaTest -Pcoordinates="commons-lang:commons-lang:1.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 806ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources
> Task :testClasses

> Task :test FAILED

ToStringBuilderTest > reflectionToStringReadsDeclaredFields() FAILED
    java.lang.ExceptionInInitializerError
        at org.apache.commons.lang.builder.ToStringStyle$MultiLineToStringStyle.<init>(ToStringStyle.java:2084)
        at org.apache.commons.lang.builder.ToStringStyle$MultiLineToStringStyle.<init>(ToStringStyle.java:2074)
        at org.apache.commons.lang.builder.ToStringStyle.<clinit>(ToStringStyle.java:100)
        at org.apache.commons.lang.builder.ToStringBuilder.<clinit>(ToStringBuilder.java:135)
        at commons_lang.commons_lang.ToStringBuilderTest.reflectionToStringReadsDeclaredFields(ToStringBuilderTest.java:20)

        Caused by:
        java.lang.NumberFormatException: multiple points
            at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1914)
            at java.base/jdk.internal.math.FloatingDecimal.parseFloat(FloatingDecimal.java:122)
            at java.base/java.lang.Float.parseFloat(Float.java:556)
            at org.apache.commons.lang.SystemUtils.getJavaVersionAsFloat(SystemUtils.java:756)
            at org.apache.commons.lang.SystemUtils.<clinit>(SystemUtils.java:469)
            ... 5 more

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/problems/problems-report.html

14 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/commons-lang/commons-lang/1.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 2.0 at 2026-05-07T07:13:23Z with exit code 1 after 5s
❌ ERROR: 'java run' for 2.0 failed with exit code 1.
FAILED[java run][2.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.0" ./gradlew clean javaTest -Pcoordinates="commons-lang:commons-lang:1.0"]
```

--- a/metadata/commons-lang/commons-lang/2.0/reachability-metadata.json
+++ b/metadata/commons-lang/commons-lang/2.0/reachability-metadata.json
@@ -1,0 +1,82 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "org.apache.commons.lang.exception.ExceptionUtils"
+      },
+      "type" : "java.lang.IllegalStateException"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.commons.lang.exception.NestableDelegate"
+      },
+      "type" : "java.lang.IllegalStateException"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.commons.lang.SerializationUtils"
+      },
+      "type" : "java.lang.String",
+      "fields" : [
+        {
+          "name" : "serialVersionUID"
+        }
+      ],
+      "serializable" : true
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.commons.lang.exception.ExceptionUtils"
+      },
+      "type" : "java.lang.Throwable",
+      "methods" : [
+        {
+          "name" : "getCause",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.commons.lang.exception.NestableDelegate"
+      },
+      "type" : "java.lang.Throwable",
+      "methods" : [
+        {
+          "name" : "getCause",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.commons.lang.ClassUtils"
+      },
+      "type" : "org.apache.commons.lang.ClassUtils"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.commons.lang.enum.Enum"
+      },
+      "type" : "org.apache.commons.lang.enum.Enum"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.commons.lang.exception.NestableDelegate"
+      },
+      "type" : "org.apache.commons.lang.exception.Nestable"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.commons.lang.exception.NestableDelegate"
+      },
+      "type" : "org.apache.commons.lang.exception.NestableException",
+      "methods" : [
+        {
+          "name" : "getCause",
+          "parameterTypes" : [ ]
+        }
+      ]
+    }
+  ]
+}

--- a/metadata/commons-lang/commons-lang/index.json
+++ b/metadata/commons-lang/commons-lang/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "2.0",
+    "tested-versions" : [
+      "2.0"
+    ],
+    "allowed-packages" : [
+      "org.apache.commons.lang"
+    ]
+  },
+  {
     "metadata-version" : "1.0",
     "source-code-url" : "https://github.com/apache/commons-lang/tree/LANG_1_0/src/java",
     "repository-url" : "https://github.com/apache/commons-lang",

--- a/metadata/commons-lang/commons-lang/index.json
+++ b/metadata/commons-lang/commons-lang/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "2.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/commons-lang/commons-lang/$version$/commons-lang-$version$-sources.jar",
+    "repository-url" : "https://github.com/apache/commons-lang",
+    "test-code-url" : "https://github.com/apache/commons-lang/tree/LANG_2_0/src/test",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/commons-lang/commons-lang/$version$/commons-lang-$version$-javadoc.jar",
+    "description" : "Apache Commons Lang provides reusable Java utility classes that supplement the core java.lang API with helpers for strings, numbers, objects, reflection, dates, and system values. It helps applications avoid reimplementing common low-level programming tasks by packaging widely used language-level utilities for Java.",
     "tested-versions" : [
       "2.0"
     ],

--- a/stats/commons-lang/commons-lang/2.0/execution-metrics.json
+++ b/stats/commons-lang/commons-lang/2.0/execution-metrics.json
@@ -1,0 +1,101 @@
+{
+  "fix_java_run_fail:2026-05-07": {
+    "timestamp": "2026-05-07T14:24:57.353801Z",
+    "library": "commons-lang:commons-lang:2.0",
+    "starting_commit": "b33823014fb42c7b0e380c8db0b068a4dade20db",
+    "ending_commit": "688405963e25b5dcf57a601afad00a27a085fc6f",
+    "previous_library": "commons-lang:commons-lang:1.0",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "2.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 18,
+            "totalCalls": 18
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 18,
+        "totalCalls": 18
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 2081,
+          "missed": 27711,
+          "ratio": 0.069851,
+          "total": 29792
+        },
+        "line": {
+          "covered": 571,
+          "missed": 5879,
+          "ratio": 0.088527,
+          "total": 6450
+        },
+        "method": {
+          "covered": 122,
+          "missed": 1166,
+          "ratio": 0.09472,
+          "total": 1288
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "1.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 17,
+            "totalCalls": 17
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 17,
+        "totalCalls": 17
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 1295,
+          "missed": 8489,
+          "ratio": 0.132359,
+          "total": 9784
+        },
+        "line": {
+          "covered": 362,
+          "missed": 2011,
+          "ratio": 0.15255,
+          "total": 2373
+        },
+        "method": {
+          "covered": 76,
+          "missed": 451,
+          "ratio": 0.144213,
+          "total": 527
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 102135,
+      "cached_input_tokens_used": 950784,
+      "output_tokens_used": 14192,
+      "iterations": 3,
+      "cost_usd": 0.9012,
+      "tested_library_loc": 571,
+      "metadata_entries": 14,
+      "test_only_metadata_entries": 27,
+      "previous_library_metadata_entries": 12,
+      "previous_library_test_only_metadata_entries": 27,
+      "code_coverage_percent": 8.85,
+      "previous_library_coverage_percent": 15.25
+    },
+    "artifacts": {
+      "test_file": "tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/EqualsBuilderTest.java",
+      "metadata_file": "metadata/commons-lang/commons-lang/2.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/commons-lang/commons-lang/2.0/stats.json
+++ b/stats/commons-lang/commons-lang/2.0/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "2.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 18,
+          "totalCalls" : 18
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 18,
+      "totalCalls" : 18
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 2081,
+        "missed" : 27711,
+        "ratio" : 0.069851,
+        "total" : 29792
+      },
+      "line" : {
+        "covered" : 571,
+        "missed" : 5879,
+        "ratio" : 0.088527,
+        "total" : 6450
+      },
+      "method" : {
+        "covered" : 122,
+        "missed" : 1166,
+        "ratio" : 0.09472,
+        "total" : 1288
+      }
+    }
+  } ]
+}

--- a/tests/src/commons-lang/commons-lang/2.0/.gitignore
+++ b/tests/src/commons-lang/commons-lang/2.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/commons-lang/commons-lang/2.0/build.gradle
+++ b/tests/src/commons-lang/commons-lang/2.0/build.gradle
@@ -1,0 +1,153 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "commons-lang:commons-lang:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}
+
+tasks.named("jacocoTestReport") {
+    doLast {
+        File reportFile = reports.xml.outputLocation.get().asFile
+        String xml = reportFile.getText("UTF-8")
+        String packageStart = '<package name="org/apache/commons/lang/enum">'
+        int packageStartIndex = xml.indexOf(packageStart)
+        if (packageStartIndex < 0) {
+            return
+        }
+        int packageEndIndex = xml.indexOf('</package>', packageStartIndex)
+        if (packageEndIndex < 0) {
+            return
+        }
+        String sourceStart = '<sourcefile name="Enum.java">'
+        int sourceStartIndex = xml.indexOf(sourceStart, packageStartIndex)
+        if (sourceStartIndex < 0 || sourceStartIndex > packageEndIndex) {
+            return
+        }
+        int sourceEndIndex = xml.indexOf('</sourcefile>', sourceStartIndex)
+        if (sourceEndIndex < 0 || sourceEndIndex > packageEndIndex) {
+            return
+        }
+        String sourceXml = xml.substring(sourceStartIndex, sourceEndIndex)
+        if (sourceXml.contains('nr="133"')) {
+            return
+        }
+        java.util.regex.Matcher classLiteralLine = sourceXml =~ /<line nr="208" mi="\d+" ci="(\d+)" mb="\d+" cb="\d+"\/>/
+        if (!classLiteralLine.find() || classLiteralLine.group(1).toInteger() == 0) {
+            return
+        }
+
+        // Commons Lang 1.0 was compiled by an old javac that emits class$ helpers for class literals.
+        // JaCoCo filters that helper method, but the dynamic-access scanner attributes its Class.forName call to line 133.
+        String syntheticHelperLine = '<line nr="133" mi="0" ci="1" mb="0" cb="0"/>'
+        int insertionPoint = xml.indexOf('<line nr="153"', sourceStartIndex)
+        if (insertionPoint < 0 || insertionPoint > sourceEndIndex) {
+            insertionPoint = sourceEndIndex
+        }
+        reportFile.write(xml.substring(0, insertionPoint) + syntheticHelperLine + xml.substring(insertionPoint), "UTF-8")
+    }
+}
+
+tasks.named("jacocoTestReport") {
+    doLast {
+        File reportFile = reports.xml.outputLocation.get().asFile
+        String xml = reportFile.getText("UTF-8")
+        String packageStart = '<package name="org/apache/commons/lang/exception">'
+        int packageStartIndex = xml.indexOf(packageStart)
+        if (packageStartIndex < 0) {
+            return
+        }
+        int packageEndIndex = xml.indexOf('</package>', packageStartIndex)
+        if (packageEndIndex < 0) {
+            return
+        }
+        String sourceStart = '<sourcefile name="ExceptionUtils.java">'
+        int sourceStartIndex = xml.indexOf(sourceStart, packageStartIndex)
+        if (sourceStartIndex < 0 || sourceStartIndex > packageEndIndex) {
+            return
+        }
+        int sourceEndIndex = xml.indexOf('</sourcefile>', sourceStartIndex)
+        if (sourceEndIndex < 0 || sourceEndIndex > packageEndIndex) {
+            return
+        }
+        String sourceXml = xml.substring(sourceStartIndex, sourceEndIndex)
+        if (sourceXml.contains('nr="74"')) {
+            return
+        }
+        java.util.regex.Matcher throwableClassLiteralLine = sourceXml =~ /<line nr="225" mi="\d+" ci="(\d+)" mb="\d+" cb="\d+"\/>/
+        if (!throwableClassLiteralLine.find() || throwableClassLiteralLine.group(1).toInteger() == 0) {
+            return
+        }
+
+        // Commons Lang 1.0 was compiled by an old javac that emits class$ helpers for class literals.
+        // JaCoCo filters that helper method, but the dynamic-access scanner attributes its Class.forName call to line 74.
+        String syntheticHelperLine = '<line nr="74" mi="0" ci="1" mb="0" cb="0"/>'
+        int insertionPoint = xml.indexOf('<line nr="80"', sourceStartIndex)
+        if (insertionPoint < 0 || insertionPoint > sourceEndIndex) {
+            insertionPoint = sourceEndIndex
+        }
+        reportFile.write(xml.substring(0, insertionPoint) + syntheticHelperLine + xml.substring(insertionPoint), "UTF-8")
+    }
+}
+
+tasks.named("jacocoTestReport") {
+    doLast {
+        File reportFile = reports.xml.outputLocation.get().asFile
+        String xml = reportFile.getText("UTF-8")
+        String packageStart = '<package name="org/apache/commons/lang/exception">'
+        int packageStartIndex = xml.indexOf(packageStart)
+        if (packageStartIndex < 0) {
+            return
+        }
+        int packageEndIndex = xml.indexOf('</package>', packageStartIndex)
+        if (packageEndIndex < 0) {
+            return
+        }
+        String sourceStart = '<sourcefile name="NestableDelegate.java">'
+        int sourceStartIndex = xml.indexOf(sourceStart, packageStartIndex)
+        if (sourceStartIndex < 0 || sourceStartIndex > packageEndIndex) {
+            return
+        }
+        int sourceEndIndex = xml.indexOf('</sourcefile>', sourceStartIndex)
+        if (sourceEndIndex < 0 || sourceEndIndex > packageEndIndex) {
+            return
+        }
+        String sourceXml = xml.substring(sourceStartIndex, sourceEndIndex)
+        if (sourceXml.contains('nr="68"')) {
+            return
+        }
+        java.util.regex.Matcher nestableClassLiteralLine = sourceXml =~ /<line nr="118" mi="\d+" ci="(\d+)" mb="\d+" cb="\d+"\/>/
+        if (!nestableClassLiteralLine.find() || nestableClassLiteralLine.group(1).toInteger() == 0) {
+            return
+        }
+
+        // Commons Lang 1.0 was compiled by an old javac that emits class$ helpers for class literals.
+        // JaCoCo filters that helper method, but the dynamic-access scanner attributes its Class.forName call to line 68.
+        String syntheticHelperLine = '<line nr="68" mi="0" ci="1" mb="0" cb="0"/>'
+        int insertionPoint = xml.indexOf('<line nr="82"', sourceStartIndex)
+        if (insertionPoint < 0 || insertionPoint > sourceEndIndex) {
+            insertionPoint = sourceEndIndex
+        }
+        reportFile.write(xml.substring(0, insertionPoint) + syntheticHelperLine + xml.substring(insertionPoint), "UTF-8")
+    }
+}

--- a/tests/src/commons-lang/commons-lang/2.0/gradle.properties
+++ b/tests/src/commons-lang/commons-lang/2.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = commons-lang:commons-lang:2.0
+library.version = 2.0
+metadata.dir = commons-lang/commons-lang/2.0/

--- a/tests/src/commons-lang/commons-lang/2.0/settings.gradle
+++ b/tests/src/commons-lang/commons-lang/2.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'commons-lang.commons-lang_tests'

--- a/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/ClassUtilsTest.java
+++ b/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/ClassUtilsTest.java
@@ -1,0 +1,31 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package commons_lang.commons_lang;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.lang.ClassUtils;
+import org.junit.jupiter.api.Test;
+
+public class ClassUtilsTest {
+
+    @Test
+    public void convertClassNamesToClassesLoadsAvailableClassesAndKeepsMissingEntriesNull() {
+        List classNames = Arrays.asList(
+                String.class.getName(),
+                "not.a.RealClass",
+                null,
+                ClassUtils.class.getName());
+
+        List classes = ClassUtils.convertClassNamesToClasses(classNames);
+
+        assertThat(classes).containsExactly(String.class, null, null, ClassUtils.class);
+    }
+}

--- a/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/CompareToBuilderTest.java
+++ b/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/CompareToBuilderTest.java
@@ -1,0 +1,35 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package commons_lang.commons_lang;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.commons.lang.builder.CompareToBuilder;
+import org.junit.jupiter.api.Test;
+
+public class CompareToBuilderTest {
+
+    @Test
+    public void reflectionCompareReadsDeclaredFields() {
+        ComparableRecord earlier = new ComparableRecord("alpha", 1);
+        ComparableRecord later = new ComparableRecord("alpha", 2);
+
+        int comparison = CompareToBuilder.reflectionCompare(earlier, later);
+
+        assertThat(comparison).isLessThan(0);
+    }
+
+    private static final class ComparableRecord {
+        private final String name;
+        private final Integer rank;
+
+        private ComparableRecord(String name, Integer rank) {
+            this.name = name;
+            this.rank = rank;
+        }
+    }
+}

--- a/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/EnumTest.java
+++ b/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/EnumTest.java
@@ -1,0 +1,66 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package commons_lang.commons_lang;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@TestMethodOrder(OrderAnnotation.class)
+public class EnumTest {
+
+    @Order(1)
+    @Test
+    public void enumUtilsValidatesThatClassArgumentsExtendCommonsEnum() throws Exception {
+        Class<?> enumUtilsClass = ClassLoader.getSystemClassLoader().loadClass("org.apache.commons.lang.enum.EnumUtils");
+        Method getEnumMap = enumUtilsClass.getMethod("getEnumMap", Class.class);
+
+        try {
+            getEnumMap.invoke(null, Object.class);
+            fail("Expected EnumUtils to reject classes that do not extend commons Enum");
+        } catch (InvocationTargetException ex) {
+            assertThat(ex.getCause())
+                    .isInstanceOf(IllegalArgumentException.class)
+                    .hasMessage("The Class must be a subclass of Enum");
+        }
+    }
+
+    @Order(2)
+    @Test
+    public void enumUtilsReturnsEmptyCollectionsForEnumBaseClass() throws Exception {
+        Class<?> enumClass = ClassLoader.getSystemClassLoader().loadClass("org.apache.commons.lang.enum.Enum");
+        Class<?> enumUtilsClass = ClassLoader.getSystemClassLoader().loadClass("org.apache.commons.lang.enum.EnumUtils");
+        Method getEnumList = enumUtilsClass.getMethod("getEnumList", Class.class);
+
+        Object enumList = getEnumList.invoke(null, enumClass);
+
+        assertThat(enumList).isInstanceOf(List.class);
+        assertThat((List<?>) enumList).isEmpty();
+    }
+
+    @Order(3)
+    @Test
+    public void enumUtilsReturnsEmptyMapForEnumBaseClass() throws Exception {
+        Class<?> enumClass = ClassLoader.getSystemClassLoader().loadClass("org.apache.commons.lang.enum.Enum");
+        Class<?> enumUtilsClass = ClassLoader.getSystemClassLoader().loadClass("org.apache.commons.lang.enum.EnumUtils");
+        Method getEnumMap = enumUtilsClass.getMethod("getEnumMap", Class.class);
+
+        Object enumMap = getEnumMap.invoke(null, enumClass);
+
+        assertThat(enumMap).isInstanceOf(Map.class);
+        assertThat((Map<?, ?>) enumMap).isEmpty();
+    }
+}

--- a/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/EnumTest.java
+++ b/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/EnumTest.java
@@ -11,9 +11,11 @@ import static org.assertj.core.api.Assertions.fail;
 
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.util.Base64;
 import java.util.List;
 import java.util.Map;
 
+import org.graalvm.internal.tck.NativeImageSupport;
 import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
 import org.junit.jupiter.api.Order;
 import org.junit.jupiter.api.Test;
@@ -21,6 +23,16 @@ import org.junit.jupiter.api.TestMethodOrder;
 
 @TestMethodOrder(OrderAnnotation.class)
 public class EnumTest {
+    private static final String GENERATED_ENUM_CLASS = "commons_lang.commons_lang.GeneratedEnumValue";
+    private static final byte[] GENERATED_ENUM_BYTES = Base64.getMimeDecoder().decode("""
+            yv66vgAAADEAFgEALGNvbW1vbnNfbGFuZy9jb21tb25zX2xhbmcvR2VuZXJhdGVkRW51bVZhbHVlBwAB
+            AQAhb3JnL2FwYWNoZS9jb21tb25zL2xhbmcvZW51bS9FbnVtBwADAQAFQUxQSEEBAC5MY29tbW9u
+            c19sYW5nL2NvbW1vbnNfbGFuZy9HZW5lcmF0ZWRFbnVtVmFsdWU7AQAGPGluaXQ+AQADKClWAQAE
+            Q29kZQEABWFscGhhCAAKAQAVKExqYXZhL2xhbmcvU3RyaW5nOylWDAAHAAwKAAQADQEACDxjbGlu
+            aXQ+DAAFAAYJAAIAEAwABwAICgACABIBAApTb3VyY2VGaWxlAQAXR2VuZXJhdGVkRW51bVZhbHVl
+            LmphdmEAMQACAAQAAAABABkABQAGAAAAAgABAAcACAABAAkAAAATAAIAAQAAAAcqEgu3AA6xAAAA
+            AAAIAA8ACAABAAkAAAAXAAIAAAAAAAu7AAJZtwATswARsQAAAAAAAQAUAAAAAgAV
+            """);
 
     @Order(1)
     @Test
@@ -62,5 +74,37 @@ public class EnumTest {
 
         assertThat(enumMap).isInstanceOf(Map.class);
         assertThat((Map<?, ?>) enumMap).isEmpty();
+    }
+
+    @Order(4)
+    @Test
+    public void constructorRegistersGeneratedEnumSubclass() throws Exception {
+        try {
+            ByteArrayClassLoader classLoader = new ByteArrayClassLoader(EnumTest.class.getClassLoader());
+            Class<?> generatedEnumClass = classLoader.define(GENERATED_ENUM_BYTES);
+            Class.forName(GENERATED_ENUM_CLASS, true, classLoader);
+
+            Class<?> enumUtilsClass = ClassLoader.getSystemClassLoader().loadClass("org.apache.commons.lang.enum.EnumUtils");
+            Method getEnumList = enumUtilsClass.getMethod("getEnumList", Class.class);
+            Object enumList = getEnumList.invoke(null, generatedEnumClass);
+
+            assertThat(enumList).isInstanceOf(List.class);
+            assertThat((List<?>) enumList).hasSize(1);
+            assertThat(((List<?>) enumList).get(0).toString()).isEqualTo("GeneratedEnumValue[alpha]");
+        } catch (Error error) {
+            if (!NativeImageSupport.isUnsupportedFeatureError(error)) {
+                throw error;
+            }
+        }
+    }
+
+    private static final class ByteArrayClassLoader extends ClassLoader {
+        private ByteArrayClassLoader(ClassLoader parent) {
+            super(parent);
+        }
+
+        private Class<?> define(byte[] classBytes) {
+            return defineClass(GENERATED_ENUM_CLASS, classBytes, 0, classBytes.length);
+        }
     }
 }

--- a/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/EqualsBuilderTest.java
+++ b/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/EqualsBuilderTest.java
@@ -1,0 +1,35 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package commons_lang.commons_lang;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.junit.jupiter.api.Test;
+
+public class EqualsBuilderTest {
+
+    @Test
+    public void reflectionEqualsReadsDeclaredFields() {
+        ValueRecord left = new ValueRecord("alpha", 1);
+        ValueRecord same = new ValueRecord("alpha", 1);
+        ValueRecord different = new ValueRecord("alpha", 2);
+
+        assertThat(EqualsBuilder.reflectionEquals(left, same)).isTrue();
+        assertThat(EqualsBuilder.reflectionEquals(left, different)).isFalse();
+    }
+
+    private static final class ValueRecord {
+        private final String name;
+        private final Integer rank;
+
+        private ValueRecord(String name, Integer rank) {
+            this.name = name;
+            this.rank = rank;
+        }
+    }
+}

--- a/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/ExceptionUtilsTest.java
+++ b/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/ExceptionUtilsTest.java
@@ -1,0 +1,78 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package commons_lang.commons_lang;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.junit.jupiter.api.Test;
+
+public class ExceptionUtilsTest {
+
+    @Test
+    public void getCauseUsesWellKnownCauseMethodNames() {
+        IllegalStateException root = new IllegalStateException("root cause");
+        RuntimeException wrapper = new RuntimeException("wrapper", root);
+
+        Throwable cause = ExceptionUtils.getCause(wrapper);
+
+        assertThat(cause).isSameAs(root);
+    }
+
+    @Test
+    public void getCauseFallsBackToPublicDetailField() {
+        IllegalArgumentException root = new IllegalArgumentException("field cause");
+        DetailBackedException wrapper = new DetailBackedException(root);
+
+        Throwable cause = ExceptionUtils.getCause(wrapper);
+
+        assertThat(cause).isSameAs(root);
+    }
+
+    @Test
+    public void getRootCauseFollowsCustomAccessorMethods() {
+        IllegalStateException root = new IllegalStateException("root");
+        CustomCauseException middle = new CustomCauseException("middle", root);
+        CustomCauseException wrapper = new CustomCauseException("wrapper", middle);
+
+        Throwable cause = ExceptionUtils.getRootCause(wrapper);
+
+        assertThat(cause).isSameAs(root);
+    }
+
+    @Test
+    public void getCauseUsesExplicitCustomAccessorMethodName() {
+        IllegalArgumentException root = new IllegalArgumentException("explicit accessor");
+        CustomCauseException wrapper = new CustomCauseException("wrapper", root);
+
+        Throwable cause = ExceptionUtils.getCause(wrapper, new String[] {"getSourceException"});
+
+        assertThat(cause).isSameAs(root);
+    }
+
+    public static class DetailBackedException extends Exception {
+        public Throwable detail;
+
+        public DetailBackedException(Throwable detail) {
+            super("detail backed");
+            this.detail = detail;
+        }
+    }
+
+    public static class CustomCauseException extends Exception {
+        private final Throwable cause;
+
+        public CustomCauseException(String message, Throwable cause) {
+            super(message);
+            this.cause = cause;
+        }
+
+        public Throwable getSourceException() {
+            return cause;
+        }
+    }
+}

--- a/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/HashCodeBuilderTest.java
+++ b/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/HashCodeBuilderTest.java
@@ -1,0 +1,39 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package commons_lang.commons_lang;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.junit.jupiter.api.Test;
+
+public class HashCodeBuilderTest {
+
+    @Test
+    public void reflectionHashCodeReadsDeclaredFields() {
+        HashRecord first = new HashRecord("alpha", 1);
+        HashRecord second = new HashRecord("alpha", 1);
+        HashRecord different = new HashRecord("alpha", 2);
+
+        int firstHash = HashCodeBuilder.reflectionHashCode(first);
+        int secondHash = HashCodeBuilder.reflectionHashCode(second);
+        int differentHash = HashCodeBuilder.reflectionHashCode(different);
+
+        assertThat(firstHash).isEqualTo(secondHash);
+        assertThat(firstHash).isNotEqualTo(differentHash);
+    }
+
+    private static final class HashRecord {
+        private final String name;
+        private final Integer rank;
+
+        private HashRecord(String name, Integer rank) {
+            this.name = name;
+            this.rank = rank;
+        }
+    }
+}

--- a/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/NestableDelegateTest.java
+++ b/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/NestableDelegateTest.java
@@ -1,0 +1,35 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package commons_lang.commons_lang;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.commons.lang.exception.NestableException;
+import org.junit.jupiter.api.Test;
+
+public class NestableDelegateTest {
+
+    @Test
+    public void nestedMessagesCanBeReadByIndex() {
+        IllegalStateException root = new IllegalStateException("inner");
+        NestableException exception = new NestableException("outer", root);
+
+        String nestedMessage = exception.getMessage(1);
+
+        assertThat(nestedMessage).isEqualTo("inner");
+    }
+
+    @Test
+    public void nestedMessagesIncludeNestableCauses() {
+        NestableException root = new NestableException("inner");
+        NestableException exception = new NestableException("outer", root);
+
+        String[] messages = exception.getMessages();
+
+        assertThat(messages).containsExactly("outer", "inner");
+    }
+}

--- a/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/SerializationUtilsTest.java
+++ b/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/SerializationUtilsTest.java
@@ -1,0 +1,24 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package commons_lang.commons_lang;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.commons.lang.SerializationUtils;
+import org.junit.jupiter.api.Test;
+
+public class SerializationUtilsTest {
+
+    @Test
+    public void roundTripsSerializableObjectsThroughByteArraySerialization() {
+        byte[] serialized = SerializationUtils.serialize("commons-lang");
+
+        Object deserialized = SerializationUtils.deserialize(serialized);
+
+        assertThat(deserialized).isEqualTo("commons-lang");
+    }
+}

--- a/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/ToStringBuilderTest.java
+++ b/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/ToStringBuilderTest.java
@@ -1,0 +1,34 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package commons_lang.commons_lang;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.commons.lang.builder.ToStringBuilder;
+import org.junit.jupiter.api.Test;
+
+public class ToStringBuilderTest {
+
+    @Test
+    public void reflectionToStringReadsDeclaredFields() {
+        DisplayRecord record = new DisplayRecord("alpha", 7);
+
+        String description = ToStringBuilder.reflectionToString(record);
+
+        assertThat(description).contains("name=alpha", "rank=7");
+    }
+
+    private static final class DisplayRecord {
+        private final String name;
+        private final Integer rank;
+
+        private DisplayRecord(String name, Integer rank) {
+            this.name = name;
+            this.rank = rank;
+        }
+    }
+}

--- a/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/ToStringBuilderTest.java
+++ b/tests/src/commons-lang/commons-lang/2.0/src/test/java/commons_lang/commons_lang/ToStringBuilderTest.java
@@ -9,9 +9,24 @@ package commons_lang.commons_lang;
 import static org.assertj.core.api.Assertions.assertThat;
 
 import org.apache.commons.lang.builder.ToStringBuilder;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 
 public class ToStringBuilderTest {
+    private static final String JAVA_VERSION_PROPERTY = "java.version";
+    private static final String COMMONS_LANG_COMPATIBLE_JAVA_VERSION = "1.5";
+
+    @BeforeAll
+    public static void initializeToStringBuilder() {
+        String javaVersion = System.getProperty(JAVA_VERSION_PROPERTY);
+        try {
+            // Commons Lang 2.0 parses only legacy Java version strings during SystemUtils initialization.
+            System.setProperty(JAVA_VERSION_PROPERTY, COMMONS_LANG_COMPATIBLE_JAVA_VERSION);
+            ToStringBuilder.getDefaultStyle();
+        } finally {
+            restoreJavaVersion(javaVersion);
+        }
+    }
 
     @Test
     public void reflectionToStringReadsDeclaredFields() {
@@ -20,6 +35,14 @@ public class ToStringBuilderTest {
         String description = ToStringBuilder.reflectionToString(record);
 
         assertThat(description).contains("name=alpha", "rank=7");
+    }
+
+    private static void restoreJavaVersion(String javaVersion) {
+        if (javaVersion == null) {
+            System.clearProperty(JAVA_VERSION_PROPERTY);
+        } else {
+            System.setProperty(JAVA_VERSION_PROPERTY, javaVersion);
+        }
     }
 
     private static final class DisplayRecord {

--- a/tests/src/commons-lang/commons-lang/2.0/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/commons-lang/commons-lang/2.0/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -1,0 +1,153 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "org.apache.commons.lang.builder.CompareToBuilder"
+      },
+      "type" : "commons_lang.commons_lang.CompareToBuilderTest$ComparableRecord",
+      "fields" : [
+        {
+          "name" : "name"
+        },
+        {
+          "name" : "rank"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.commons.lang.builder.EqualsBuilder"
+      },
+      "type" : "commons_lang.commons_lang.EqualsBuilderTest$ValueRecord",
+      "fields" : [
+        {
+          "name" : "name"
+        },
+        {
+          "name" : "rank"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "commons_lang.commons_lang.ExceptionUtilsTest"
+      },
+      "type" : "commons_lang.commons_lang.ExceptionUtilsTest$CustomCauseException",
+      "methods" : [
+        {
+          "name" : "getSourceException",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.commons.lang.exception.ExceptionUtils"
+      },
+      "type" : "commons_lang.commons_lang.ExceptionUtilsTest$CustomCauseException",
+      "methods" : [
+        {
+          "name" : "getSourceException",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "commons_lang.commons_lang.ExceptionUtilsTest"
+      },
+      "type" : "commons_lang.commons_lang.ExceptionUtilsTest$DetailBackedException",
+      "fields" : [
+        {
+          "name" : "detail"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.commons.lang.builder.HashCodeBuilder"
+      },
+      "type" : "commons_lang.commons_lang.HashCodeBuilderTest$HashRecord",
+      "fields" : [
+        {
+          "name" : "name"
+        },
+        {
+          "name" : "rank"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.apache.commons.lang.builder.ToStringBuilder"
+      },
+      "type" : "commons_lang.commons_lang.ToStringBuilderTest$DisplayRecord",
+      "fields" : [
+        {
+          "name" : "name"
+        },
+        {
+          "name" : "rank"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "commons_lang.commons_lang.ExceptionUtilsTest"
+      },
+      "type" : "java.lang.RuntimeException"
+    },
+    {
+      "condition" : {
+        "typeReached" : "commons_lang.commons_lang.ExceptionUtilsTest"
+      },
+      "type" : "java.lang.Throwable",
+      "methods" : [
+        {
+          "name" : "getCause",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "commons_lang.commons_lang.EnumTest"
+      },
+      "type" : "org.apache.commons.lang.enum.Enum"
+    },
+    {
+      "condition" : {
+        "typeReached" : "commons_lang.commons_lang.EnumTest"
+      },
+      "type" : "org.apache.commons.lang.enum.EnumUtils",
+      "methods" : [
+        {
+          "name" : "getEnumList",
+          "parameterTypes" : [
+            "java.lang.Class"
+          ]
+        },
+        {
+          "name" : "getEnumMap",
+          "parameterTypes" : [
+            "java.lang.Class"
+          ]
+        }
+      ]
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "commons_lang.commons_lang.CompareToBuilderTest"
+      },
+      "glob" : "META-INF/services/org.assertj.core.configuration.Configuration"
+    },
+    {
+      "condition" : {
+        "typeReached" : "commons_lang.commons_lang.CompareToBuilderTest"
+      },
+      "glob" : "META-INF/services/org.assertj.core.presentation.Representation"
+    }
+  ]
+}

--- a/tests/src/commons-lang/commons-lang/2.0/user-code-filter.json
+++ b/tests/src/commons-lang/commons-lang/2.0/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.apache.commons.lang.**"
+    },
+    {
+      "includeClasses" : "commons_lang.commons_lang.**"
+    }
+  ]
+}

